ui/app/src/utils/hooks.ts

```diff
@@ -155,3 +157,15 @@ export function useStateResource<ReturnType = any, SourceType = GlobalState>(
 export function useOnMount(componentDidMount: EffectCallback) {
   useEffect(componentDidMount, []);
 }
+
+export function useDebouncedInput(observer: (keywords: string) => any) {
```

Comment:
- add debounceTime param
- default it to 250 (ms)
- use it for debounce
- add to effect deps